tests/adapters/compliance_tests/set_interface_state_test.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2015 Internap.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from hamcrest import assert_that, is_, contains_string
+
+from netman.core.objects.exceptions import UnknownInterface
+from netman.core.objects.interface_states import ON, OFF
+from tests.adapters.configured_test_case import ConfiguredTestCase
+
+
+class SetInterfaceStateTest(ConfiguredTestCase):
+    _dev_sample = "juniper"
+
+    def test_set_interface_state_on(self):
+        self.try_to.set_interface_state(self.test_ports[0].name, state=ON)
```

Comment:
if set_interface_state is what you test, do NOT use try_to, this will hide NotImplementedError that are core to skipping or not theses tests